metis/lib/server/controllers/upload_controller.rb

```diff
@@ -58,6 +52,21 @@ def upload
 
   private
 
+  def create_upload(params)
+    Metis::Upload.find_or_create(
+      file_name: params[:file_path],
+      bucket: params[:bucket],
+      metis_uid: metis_uid,
+      project_name: params[:project_name]
+    ) do |f|
+      f.author = Metis::File.author(params[:user])
+      f.file_size = 0
+      f.current_byte_position = 0
+      f.next_blob_size = -1
+      f.next_blob_hash = ''
+    end
+  end
+
```

Comment:
If this has to move out of the main controller method I'd prefer to keep methods like this on the relevant model, like `Metis::Upload.fetch` or something, rather than adrift in the controller class (where it becomes confusing who they are "helping").